unified-runtime/source/adapters/level_zero/v2/queue_api.cpp

```diff
@@ -348,19 +348,21 @@ ur_result_t urEnqueueUSMDeviceAllocExp(
     uint32_t numEventsInWaitList, const ur_event_handle_t *phEventWaitList,
```

Comment:
This file is auto-generated. If you want to modify it you need to change scripts/templates/queue_api.cpp.mako (same is true for the corresponding .hpp file)

Reply:
But I don't think you need to actually modify it. As far as I understand you wanted to add ur_queue_handle_t param so that you can pass it to EnqueuedPool functions (like getBestFit, etc.). However, if I'm not mistaken, the queue is only used as an identifier by the EnqueuedPool implementation. This means that you can just do something like this in queue_immediate_in_order.cpp:

```
...
getBestFit(size, alignment, reinterpret_cast<ur_queue_handle_t>(this));
```

If the EnqueuedPool actually needs to have valid queue handle, then I think it would be better to rewrite EnqueuedPool  (which you might need to do anyway, as it has some logic for handling events that is legacy-specific).